jbs/src/main/java/org/openjdk/skara/jbs/JdkVersion.java

```diff
@@ -56,6 +56,10 @@ private static List<String> splitComponents(String raw) {
                 if (legacyMatcher.group(3) != null) {
                     finalComponents.add(legacyMatcher.group(3));
                 }
+                if (legacyMatcher.groupCount() > 3 && legacyMatcher.group(4) != null) {
```

Comment:
Should that be `> 4`? if it was equal to 4 wouldn't `group(4)` be OOB?

Reply:
Group 0 is the complete expression and is not included in the count by groupCount(). Perhaps `>= 4` is more readable as the number 4 is the relevant one.